openfe/protocols/openmm_rfe/_rfe_utils/relative.py

```diff
@@ -2346,21 +2347,99 @@ def _create_mdtraj_topology(self):
             at1_uniq = at1_hybrid_idx in self._atom_classes['unique_new_atoms']
             at2_uniq = at2_hybrid_idx in self._atom_classes['unique_new_atoms']
             if at1_uniq or at2_uniq:
-                if atom1.index in self._atom_classes['unique_new_atoms']:
-                    atom1_to_bond = added_atoms[atom1.index]
+                if at1_uniq:
+                    atom1_to_bond = added_atoms[at1_hybrid_idx]
                 else:
-                    atom1_to_bond = atom1
+                    old_idx = self._hybrid_to_old_map[at1_hybrid_idx]
+                    atom1_to_bond = hybrid_topology.atom(old_idx)
 
-                if atom2.index in self._atom_classes['unique_new_atoms']:
-                    atom2_to_bond = added_atoms[atom2.index]
+                if at2_uniq:
+                    atom2_to_bond = added_atoms[at2_hybrid_idx]
                 else:
-                    atom2_to_bond = atom2
+                    old_idx = self._hybrid_to_old_map[at2_hybrid_idx]
+                    atom2_to_bond = hybrid_topology.atom(old_idx)
 
                 hybrid_topology.add_bond(atom1_to_bond, atom2_to_bond)
 
         return hybrid_topology
 
 
+    def _create_hybrid_topology(self):
+        """
+        Create a hybrid openmm.app.Topology from the input old and new
+        Topologies.
+
+        Note
+        ----
+        This is not intended for parameterisation purposes, but instead
+        for system visualisation.
+        """
+
+        hybrid_top = app.Topology()
+
+        # In the first instance, create a list of necessary atoms from
+        # both old & new Topologies
+        atom_list = []
+
+        for pidx in range(self.hybrid_system.getNumParticles()):
+            if pidx in self._hybrid_to_old_map:
+                idx = self._hybrid_to_old_map[pidx]
+                atom_list.append(list(self._old_topology.atoms())[idx])
+            else:
+                idx = self._hybrid_to_new_map[pidx]
+                atom_list.append(list(self._new_topology.atoms())[idx])
+
+        # Now we loop over the atoms and add them in alongside chains & resids
+        
+        # Non ideal variables to track the previous set of residues & chains
+        # without having to constantly search backwards
+        prev_res = None
```

Comment:
This is ugly, I just ran out of steam on how to do this without lots of complexity below. I'm sure it's a simple thing I'm not thinking of.